### ISSUE TITLE
Android background location tracking automatically stops after a few seconds

### ISSUE THREAD

**Issue by andreirat** [transistorsoft/react-native-background-geolocation#1870]
## Your Environment
* Plugin version: 
* Platform: Android
* OS version: 13
* Device manufacturer / model: Samsung S21 Ultra
* React Native version (`react-native -v`): 0.72.6
* Plugin config
```javascript <!-- Syntax-highlighting:  paste your code below -->
 await BackgroundGeolocation.ready({
        distanceFilter: Platform.OS === 'android' ? 0 : 1000,
        locationUpdateInterval: 5,
        stopTimeout: 5,
        debug: true,
        disableLocationAuthorizationAlert: true,
        logLevel: BackgroundGeolocation.LOG_LEVEL_VERBOSE,
        desiredAccuracy: BackgroundGeolocation.DESIRED_ACCURACY_HIGH,
        stopOnTerminate: false, // <-- Allow the background-service to continue tracking when user closes the app.
        startOnBoot: true,
        batchSync: false,
        autoSync: true,
        enableHeadless: Platform.OS === 'android',
      }).then((state) => console.log('INDEX STATE -----', state));
```

## Expected Behavior
The background service should not close after a few seconds. 

## Actual Behavior
The background service starts when calling `BackgroundGeolocation.start()` but it automatically stops after 2 seconds and the `onLocation` callback is not triggered. 

## Steps to Reproduce

1. Initialize the library
2. Call `BackgroundGeolocation.start()`;
## Debug logs

<details><summary>Logs</summary>

``` <!-- Syntax highlighting:  DO NOT REMOVE -->

```

</details>

**Comment by christocracy** [comment]
> Plugin version:

> The background service should not close after a few seconds.

That's exactly what it's supposed to do.  Go read the wiki "Philosophy of Operation" then go outside for a nice long walk (>1km) and listen to the debug sound fx.

See api docs Config.debug to learn what the debug sound fx mean.

**Comment by andreirat** [comment]
Thanks @christocracy! I needed to test while in dev mode with the phone connected to my laptop. I was able to do it with Lockito and by disabling battery optimisation for my app. 

Do you know if there is a way to avoid that? Should we ask our users to do that manually?

**Comment by christocracy** [comment]
Yes, user's need to manually adjust settings according to https://dontkillmyapp.com.  This is the fault of Android device manufacturers violating the Android api after years of developers abusing devices with aggressive code.

See api docs "DeviceSettings"

**Comment by andreirat** [comment]
Thank you for your help.
